### PR TITLE
Utility to dump/restore ledger metadata to/from file

### LINKED ISSUE #1604: Default ledger id formatter should be LongLedgerIdFormatter
Currently the default ledger id formatter in bookie shell is set to UUID formatter. I think that is only applicable when using LedgerAdv. so I think we need to change the default ledger id formatter to long ledger id formatter.

### PR DESCRIPTION
**Pull request by ivankelly**
Dumping will dump the serialized metadata to a file. Restoring will
parse the metadata from the file and write it to the metadata store.

Also changes the LedgerIdFormatter to default to long formatter (it
was UUID before, and we don't even currently support UUID ledger ids).

This fixes #1604

### CODE REVIEW

**Comment by sijie** [comment]
@ivankelly I linked this PR with #1604, since it fixes it

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java [hunk 1]
```diff
@@ -1115,6 +1095,8 @@ public void operationComplete(int rc, LedgerMetadata result) {
         LedgerMetadataCmd() {
             super(CMD_LEDGERMETADATA);
             lOpts.addOption("l", "ledgerid", true, "Ledger ID");
+            lOpts.addOption("dumptofile", true, "Dump metadata for ledger, to a file");
```
Comment (sijie) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java]: shall we add validation that `dumptofile` and `restorefromfile` are mutually exclusive

Comment (ivankelly) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java]: added

**Comment by ivankelly** [comment]
rerun integration tests
rerun bookkeeper-server replication tests

**Pull request merged**